differ/differ_test.go

```diff
@@ -1451,3 +1453,118 @@ func TestProcessClustersWeeklyDigest(t *testing.T) {
+		if err != nil {
+			log.Fatal().Msg(err.Error())
+		}
+	}))
+	defer server.Close()
+
+	config := conf.ConfigStruct{
+		ServiceLog: conf.ServiceLogConfiguration{
+			Enabled:            true,
+			TotalRiskThreshold: 1,
+			EventFilter:        "totalRisk > totalRiskThreshold",
+		},
+		Dependencies: conf.DependenciesConfiguration{
+			TemplateRendererServer:   server.URL,
+			TemplateRendererEndpoint: "/rendered_reports",
+			TemplateRendererURL:      server.URL + "/rendered_reports",
+		},
+	}
+	serviceLogEventThresholds.TotalRisk = 1
+	serviceLogEventFilter = "totalRisk > totalRiskThreshold"
+
+	errorKeys := map[string]utypes.RuleErrorKeyContent{
+		"RULE_1": {
+			Metadata: utypes.ErrorKeyMetadata{
+				Description: "This reason is more than 255 characters long. This summary is more than 255 characters long. This summary is more than 255 characters long. This summary is more than 255 characters long. This summary is more than 255 characters long. This summary is more than 255 characters long.",
+				Impact: utypes.Impact{
+					Name:   "impact_1",
+					Impact: 3,
+				},
+				Likelihood: 2,
+			},
+			Reason:    "rule 1 reason",
+			HasReason: true,
+		},
+		"RULE_2": {
+			Metadata: utypes.ErrorKeyMetadata{
+				Description: "rule 2 error key description",
+				Impact: utypes.Impact{
+					Name:   "impact_2",
+					Impact: 2,
+				},
+				Likelihood: 3,
+			},
+			HasReason: false,
+		},
+	}
+
+	ruleContent := types.RulesMap{
+		"rule_1": {
+			Summary:    "rule 1 summary",
+			Reason:     "This reason is more than 255 characters long. This summary is more than 255 characters long. This summary is more than 255 characters long. This summary is more than 255 characters long. This summary is more than 255 characters long. This summary is more than 255 characters long.",
+			Resolution: "rule 1 resolution",
+			MoreInfo:   "rule 1 more info",
+			ErrorKeys:  errorKeys,
+			HasReason:  true,
+		},
+		"rule_2": {
+			Summary:    "rule 2 summary",
+			Reason:     "rule 2 reason",
+			Resolution: "rule 2 resolution",
+			MoreInfo:   "rule 2 more info",
+			ErrorKeys:  errorKeys,
+			HasReason:  false,
+		},
+	}
+
+	cluster := types.ClusterEntry{
+		OrgID:         1,
+		AccountNumber: 1,
+		ClusterName:   "first_cluster",
+		KafkaOffset:   0,
+		UpdatedAt:     types.Timestamp(testTimestamp),
+	}
+
+	var deserialized types.Report
+	reportsJSON := types.ClusterReport("{\"analysis_metadata\":{\"metadata\":\"some metadata\"},\"reports\":[{\"rule_id\":\"rule_1|RULE_1\",\"component\":\"ccx_rules_ocp.external.rules.rule_1.report\",\"type\":\"rule\",\"key\":\"RULE_1\",\"details\":\"some details\"},{\"rule_id\":\"rule_2|RULE_2\",\"component\":\"ccx_rules_ocp.external.rules.rule_2.report\",\"type\":\"rule\",\"key\":\"RULE_2\",\"details\":\"some details\"}]}")
```

Comment:
`ClusterReport` test data can also be found in the test data repo if needed

Reply:
Replaced, thanks for suggestion :)